content/blog/2019/corda-prevent-spend-of-broadcasted-transaction/index.md

```diff
@@ -0,0 +1,145 @@
+
+These rules are a good basis to build up your own contract's validation. Explicitly stating which parties must sign a transaction is an important requirement in proving its validity. Further rules can be added as needed.
+
+For example, the following checks could be added to the example above:
+
+```kotlin
+override fun verify(tx: LedgerTransaction) {
+  val commandWithParties: CommandWithParties<Commands> = tx.commands.requireSingleCommand()
+  val command = commandWithParties.value
+  when (command) {
+    is Commands.Send -> // validation for sending
+    is Commands.Reply -> requireThat {
+      // general requirements from previous snippet
+      // precise requirements for `MessageState`
+      "One input should be consumed when replying to a message." using (tx.inputs.size == 1)
+      "Only one output state should be created when replying to a message." using (tx.outputs.size == 1)
+      val output = tx.outputsOfType<MessageState>().single()
+      val input = tx.inputsOfType<MessageState>().single()
+      "Only the original message's recipient can reply to the message" using (output.sender == input.recipient)
+      "The reply must be sent to the original sender" using (output.recipient == input.sender)
+      // covered by the general requirements
+      "The original sender must be included in the required signers" using commandWithParties.signers.contains(input.sender.owningKey)
+      "The original recipient must be included in the required signers" using commandWithParties.signers.contains(input.recipient.owningKey)
+    }
+  }
+}
+```
+
+These rules are specific to the `MessageState`'s use-case. Restricting the parties who can reply to a message. More precisely, the sender of the reply must be the recipient of the original message and can only be sent back to the original sender. These rules make sense in this scenario.
+
+Even with these rules, dodgy transactions can still go through. The contract checks the signatures based on the transaction's states as well as the contents of the states. But, there are still some ways around these.
+
+That being said, some of the situations can only occur if transactions have been sent to parties not originally involved in the exchange. I briefly mentioned this in my previous post, [Broadcasting a transaction to external organisations](https://lankydan.dev/broadcasting-a-transaction-to-external-organisations). Broadcasting transactions is what this whole post is predicated around. It might be worth rereading the title as a refresher since I haven't touched on the topic much in this section.
```

Comment:
"Broadcasting transactions is what this whole post is predicated around." maybe mention that at the top. "In my last post XXX i talked about YYY, this carries on from that, focusing specifically on ZZZ"